### PR TITLE
Hide Forum in SaaS

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

PR in deploy repo: https://github.com/3scale/deploy/pull/727

Updates public forum support helper to check rolling update, not only the provider setting for Forum. This way by switching the rolling update `forum` off all Forum functionality will be hidden.

New providers won't have Forum templates in their CMS either.

ForumsController will now raise an error when the RU is off, showing a 404 page:
Screenshot 2022-01-24 at 18 33 43

**Which issue(s) this PR fixes** 

[THREESCALE-8032: Hide forum functionality from all users (dev portal)](https://issues.redhat.com/browse/THREESCALE-8032)

**Verification steps** 

Set rolling update `forum: false` and reset the DB, then start the server. Forum should not be visible anywhere:
- Audience > Forum
- Audience > Dev portal > Settings > Forum settings
- Audience > Dev portal > Content (search for Forum templates)
- Manually access `/forum`
- Visit to Dev portal and navigate to /forum (should return Page not found)

### CODE REVIEW

**Review by akostadinov — DISMISSED** [review]

**Review by nidhi-soni1104 — DISMISSED** [review]
LGTM

**Review by akostadinov — APPROVED** [review]

File: features/old/forums/privacy.feature [hunk 1]
```diff
@@ -9,14 +9,32 @@ Feature: Forum privacy
...
-    Given the forum of provider "foo.3scale.localhost" is public
-    When I am not logged in
+    Scenario: Public forum does not require a logged in user
+      Given the forum of provider "foo.3scale.localhost" is public
+      When I am not logged in
       And I go to the forum page
-    Then I should be on the forum page
+      Then I should be on the forum page
+
+  Rule: Forum is disabled
```
Comment (akostadinov) [inline features/old/forums/privacy.feature:28]: btw why should we switch to this syntax?

**Review by akostadinov — DISMISSED** [review]
looks nice as a temporary measure before we delete everything, thanks

**Review by nidhi-soni1104 — DISMISSED** [review]

File: features/old/forums/privacy.feature (hunk 1 above)
Comment (josemigallas) [reply features/old/forums/privacy.feature:28]: AFAIK this is the only way to branch a feature file with different Background blocks. It's just a way to override previews Background with minor changes, in this case, disabling forums.

**Comment by akostadinov** [comment]
If failing tests are forum related, I think you can just mark them as `@wip` and not invest time into fixing them if they go away anyway.

**Review by akostadinov — APPROVED** [review]

**Pull request merged**